### PR TITLE
Add mailbox log parsing support

### PR DESCRIPTION
**Pull request by brian-mckenzie**
Add mailbox log parsing support to the Switchtec library file. Mailbox logs are different from app logs as follows:
-the log definition file doesn't divide the definitions into modules. All definitions are in a single list.
-log severities are always the same
-bit 27 of the 3rd DWord of a log entry indicates the log source (BL1/BL2)

The "log-parse" command now takes a parameter to specify the log type (APP or MAILBOX) so that mailbox logs can be parsed.

### CODE REVIEW

File: lib/switchtec.c [hunk 1]
```diff
@@ -774,7 +768,7 @@ static int read_log_defs(FILE *log_def_file, struct log_defs *defs)
 		tok = strtok(NULL, " \t");
 		if (!tok)
 			continue;
-        	mod_id = strtol(tok, NULL, 0);
+		mod_id = strtol(tok, NULL, 0);
```
Comment (lsgunth) [inline lib/switchtec.c:771]: Always do whitespace cleanup in another patch.

File: lib/switchtec.c [hunk 2]
```diff
@@ -789,40 +783,121 @@ static int read_log_defs(FILE *log_def_file, struct log_defs *defs)
...
+ * @return 0 on success, negative value on failure
+ */
+static int read_log_defs(FILE *log_def_file,
+			 enum switchtec_log_parse_type log_type,
+			 struct log_defs *defs)
+{
+	int ret;
+
+	/* allocate some log definition entries */
+	ret = realloc_log_defs(defs, 200);
```
Comment (lsgunth) [inline lib/switchtec.c:888]: This is confusing me.... read_mailbox_log_defs() never seems to call realloc_log_defs() so presumably it's only actually using one module... But we always allocate 200?

I don't really see any value in this static function, it's probably better to just put the allocation into each of read_app_log_defs() and read_mailbox_log_defs() respectively... they clearly need different things.

File: lib/switchtec.c [hunk 3]
```diff
@@ -880,26 +964,45 @@ static int write_parsed_log(struct log_a_data log_data[],
...
-
-		/* validate the module ID */
-		if ((mod_id > defs->num_alloc) ||
-		    (strlen(defs->module_defs[mod_id].mod_name) == 0)) {
-			if (fprintf(log_file, "(Invalid module ID: 0x%x)\n",
-				    mod_id) < 0)
-				goto ret_print_error;
-			continue;
+		if (log_type == SWITCHTEC_LOG_PARSE_TYPE_APP) {
+			/*
```
Comment (lsgunth) [inline lib/switchtec.c:968]: So why is the functionality changing in this branch? That's not expected. If you want to change the functionality of APP logs, it should be done first in another patch with a commit message that tells me why we are making that change.

File: lib/switchtec.c [hunk 4]
```diff
@@ -880,26 +964,45 @@ static int write_parsed_log(struct log_a_data log_data[],
...
+			 * app log: module ID and log severity are in the 3rd
+			 * DWord
+			 */
+			mod_id = (log_data[i].data[2] >> 16) & 0xFFF;
+			log_sev = (log_data[i].data[2] >> 28) & 0xF;		
+
+			if ((mod_id > defs->num_alloc) ||
+			    (defs->module_defs[mod_id].mod_name == NULL) ||
```
Comment (lsgunth) [inline lib/switchtec.c:976]: So why is the functionality changing here? That's not expected. If you want to change the functionality of APP logs, it should be done first in another patch with a commit message that tells me why we are making that change.

File: lib/switchtec.c [hunk 5]
```diff
@@ -960,7 +1068,7 @@ static int log_a_to_file(struct switchtec_dev *dev, int sub_cmd_id,
 
 	if (log_def_file != NULL) {
 		/* read the log definition file into defs */
-		ret = read_log_defs(log_def_file, &defs);
+		ret = read_log_defs(log_def_file, true, &defs);
```
Comment (lsgunth) [inline lib/switchtec.c:1071]: Not withstanding that I think we should drop this function; but using "true" here is very confusing.

File: lib/switchtec.c [hunk 6]
```diff
@@ -986,7 +1094,8 @@ static int log_a_to_file(struct switchtec_dev *dev, int sub_cmd_id,
 
 			/* parse the log data and write it to a file */
 			ret = write_parsed_log(res.data, res.hdr.count,
-					       entry_idx, &defs, log_file);
+					       entry_idx, &defs, true,
```
Comment (lsgunth) [inline lib/switchtec.c:1097]: Don't use true/false when the argument takes an enum. Use actual enum values.

File: cli/main.c [hunk 7]
```diff
@@ -944,29 +944,39 @@ static int log_dump(int argc, char **argv)
...
+		{"MAILBOX", SWITCHTEC_LOG_PARSE_TYPE_MAILBOX, "mailbox log"},
+		{}
+	};
+
 	static struct {
-		struct switchtec_dev *dev;
-		int out_fd;
-		const char *out_filename;
-		unsigned type;
+		enum switchtec_log_parse_type log_type;
```
Comment (lsgunth) [inline cli/main.c:960]: Cleanup should happen in it's own commit. It's confusing, but I assume the fields being removed were never used in the first place?

Comment (brian-mckenzie) [reply cli/main.c:960]: Right, they were never used. I'll remove them in a separate commit.

File: lib/switchtec.c (hunk 1 above)
Comment (brian-mckenzie) [reply lib/switchtec.c:771]: Sure.

File: lib/switchtec.c (hunk 2 above)
Comment (brian-mckenzie) [reply lib/switchtec.c:888]: Sure. I'll remove this function and move the allocation into those ones.

File: lib/switchtec.c (hunk 3 above)
Comment (brian-mckenzie) [reply lib/switchtec.c:968]: The functionality isn't changing for app logs. The parsing is just reorganized since some of it doesn't apply to mailbox logs.

File: lib/switchtec.c (hunk 4 above)
Comment (brian-mckenzie) [reply lib/switchtec.c:976]: Same as above.

File: lib/switchtec.c (hunk 5 above)
Comment (brian-mckenzie) [reply lib/switchtec.c:1071]: Right -- originally the parameter was a bool, but I updated it to the enum and forgot to change this.

File: lib/switchtec.c (hunk 6 above)
Comment (brian-mckenzie) [reply lib/switchtec.c:1097]: Same as the above -- I forgot to update it to the enum.

File: cli/argconfig.c [hunk 8]
```diff
@@ -470,6 +470,23 @@ static int cfg_long_suffix_handler(const char *optarg, void *value_addr,
 	return 0;
 }
 
+static int cfg_long_long_handler(const char *optarg, void *value_addr,
```
Comment (brian-mckenzie) [inline cli/argconfig.c:473]: Not sure why these unrelated commits showed up -- I'll resolve this.

Comment (brian-mckenzie) [reply cli/argconfig.c:473]: I'm cancelling this PR. Please use this one instead: https://github.com/Microsemi/switchtec-user/pull/226
